compiler/grainformat/dune

```
@@ -0,0 +1,43 @@
+(library
+ (name reformat)
+ (public_name grainformat.reformat)
+ (modules reformat res_doc res_minibuffer walktree debug)
+ (libraries grain grain_parsing grain_utils))
```

I believe for completeness sake, we'll need to add `grain_diagnostics` here
```suggestion
 (libraries grain grain_diagnostics grain_parsing grain_utils))
```